### PR TITLE
Correctly bookeep threads when we reply on matrix

### PR DESCRIPTION
**Pull request by Cadair**
This is the ts port of #174

### CODE REVIEW

File: src/BridgedRoom.ts [hunk 1]
```diff
@@ -379,9 +379,10 @@ export class BridgedRoom {
         };
 
         const reply = await this.findParentReply(message);
+        let parentStoredEvent;
```
Comment (Cadair) [inline src/BridgedRoom.ts]: I figured out that this has the type `EventEntry` but I don't know how to reference that type (linter was saying it was undefined)

Comment (Half-Shot) [reply src/BridgedRoom.ts]:
```suggestion
        let parentStoredEvent: EventEntry;
```

also import `import { EventEntry } from "./datastore/Models";`

Comment (Cadair) [reply src/BridgedRoom.ts]: Having done this my linter tells me, `Type EventEntry | null is not assignable to type EventEntry` on the line below. What do I do to allow it to do both?

**Comment by Half-Shot** [comment]
There appears to be other changes in here too, but I'll allow it.

**Pull request merged**